pkg/deprecatetemplates/allowlist.go

```diff
@@ -0,0 +1,150 @@
+package deprecatetemplates
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+
+	jc "github.com/openshift/ci-tools/pkg/jobconfig"
+)
+
+const (
+	presubmit  = "presubmit"
+	postsubmit = "postsubmit"
+	periodic   = "periodic"
+	release    = "release"
+	unknown    = "unknown"
+)
+
+func isGenerated(job config.JobBase) bool {
```

Comment:
we have private copies of this function in a number of places now in this repo - can we DRY it out?